tests/integration/ha_tests/helpers.py

```diff
@@ -334,3 +345,150 @@ async def get_units_hostnames(ops_test: OpsTest) -> List[str]:
+        "-m1",
+        '"Starting an election due to step up request"',
+    )
+
+    # Gets a stream of mongod container logs from kubectl and pipes them through grep looking for a
+    # step down election messages. The check is successful if one of the greps finds a match, the
+    # rest should be terminated after a reasonable wait. All the logs should be checked since any
+    # node can emit the log.
+    procs = []
+    for unit in ops_test.model.applications[APP_NAME].units:
+        kubectl_cmd[5] = unit.name.replace("/", "-")
+        kubectl_proc = subprocess.Popen(kubectl_cmd, stdout=subprocess.PIPE)
+        grep_proc = subprocess.Popen(
+            grep_cmd, stdin=kubectl_proc.stdout, stdout=subprocess.DEVNULL
+        )
+        procs.append((kubectl_proc, grep_proc))
+
+    # Wait on the first grep pipe to potentially finish successfully. If it does not, don't wait
+    # for the rest. The check is successful if any of the greps manage to find the step down
+    # message.
+    timeout = 30
+    success = False
+    for kubectl_proc, grep_proc in procs:
+        try:
+            grep_proc.communicate(timeout=timeout)
+        except subprocess.TimeoutExpired:
+            pass
+        if grep_proc.poll() == 0:
+            success = True
+        grep_proc.terminate()
+        kubectl_proc.terminate()
+        timeout = 0
+
+    assert success, "old primary departed without stepping down."
+
+
+async def set_log_level(ops_test: OpsTest, level: int, component: str = None) -> None:
+    """Sets a given loglevel for a given component for each mongodb unit."""
+    pass_unit = ops_test.model.applications[APP_NAME].units[0].name
+    cmd = [
+        "ssh",
+        "--container",
+        MONGODB_CONTAINER_NAME,
+        "",
+        "mongosh",
+        "-u",
+        "operator",
+        "-p",
+        await get_password(ops_test, int(pass_unit.split("/")[1])),
+        "--quiet",
+        "--eval",
+        "",
+    ]
+
+    awaits = []
+    for unit in ops_test.model.applications[APP_NAME].units:
+        cmd[3] = unit.name
+        cmd[-1] = cmd[-1] = f"\"db.setLogLevel({level}, '{component}')\""
+        awaits.append(ops_test.juju(*cmd))
+    await gather(*awaits)
+
+
+async def get_total_writes(ops_test: OpsTest) -> int:
+    """Gets the total writes from the test application action."""
+    application_name = await get_application_name(ops_test, "application")
+    application_unit = ops_test.model.applications[application_name].units[0]
+    stop_writes_action = await application_unit.run_action("stop-continuous-writes")
+    await stop_writes_action.wait()
+    total_expected_writes = int(stop_writes_action.results["writes"])
+    assert total_expected_writes > 0, "error while getting total writes."
+    return total_expected_writes
+
+
+async def kubectl_delete(ops_test: OpsTest, unit: ops.model.Unit, wait: bool = True) -> None:
+    """Delete the underlying pod for a unit."""
+    kubectl_cmd = (
+        "microk8s",
```

Comment:
this test may fail if run on a non microk8s environment. instead, i would consider using the `kubectl` command only (and running `microk8s config > ~/.kube/config` locally)